safe_transaction_service/history/services/transaction_service.py

```diff
@@ -276,57 +280,60 @@ def get_all_tx_hashes(
         """
         Now that we know how to paginate, we retrieve the real transactions
 
         :param safe_address:
-        :param hashes_to_search:
+        :param ids_to_search: `SafeTxHash` for MultisigTransactions, `txHash` for other transactions
         :return:
         """
         cached_txs = {
-            hash_to_search: cached_tx
-            for hash_to_search, cached_tx in zip(
-                hashes_to_search,
-                self.get_txs_from_cache(safe_address, hashes_to_search),
+            id_to_search: cached_tx
+            for id_to_search, cached_tx in zip(
+                ids_to_search,
+                self.get_txs_from_cache(safe_address, ids_to_search),
             )
             if cached_tx
         }
-        hashes_not_cached = [
+        id_not_cached = [
             hash_to_search
-            for hash_to_search in hashes_to_search
+            for hash_to_search in ids_to_search
             if hash_to_search not in cached_txs
         ]
-        multisig_txs = {
-            multisig_tx.safe_tx_hash: multisig_tx
+        id_with_multisig_txs: Dict[HexStr, List[MultisigTransaction]] = {
+            multisig_tx.safe_tx_hash: [multisig_tx]
             for multisig_tx in MultisigTransaction.objects.filter(
-                safe=safe_address, safe_tx_hash__in=hashes_not_cached
+                safe=safe_address, safe_tx_hash__in=id_not_cached
             )
             .with_confirmations_required()
             .prefetch_related("confirmations")
             .select_related("ethereum_tx__block")
             .order_by("-nonce", "-created")
         }
 
-        module_txs = {
-            module_tx.internal_tx.ethereum_tx_id: module_tx
-            for module_tx in ModuleTransaction.objects.filter(
-                safe=safe_address, internal_tx__ethereum_tx__in=hashes_not_cached
-            ).select_related("internal_tx")
-        }
+        id_with_module_txs: Dict[HexStr, List[ModuleTransaction]] = {}
+        for module_tx in ModuleTransaction.objects.filter(
+            safe=safe_address, internal_tx__ethereum_tx__in=id_not_cached
+        ).select_related("internal_tx"):
+            id_with_module_txs.setdefault(
+                module_tx.internal_tx.ethereum_tx_id, []
+            ).append(module_tx)
 
-        plain_ethereum_txs = {
-            ethereum_tx.tx_hash: ethereum_tx
+        id_with_plain_ethereum_txs: Dict[HexStr, List[EthereumTx]] = {
+            ethereum_tx.tx_hash: [ethereum_tx]
             for ethereum_tx in EthereumTx.objects.filter(
-                tx_hash__in=hashes_not_cached
+                tx_hash__in=id_not_cached
             ).select_related("block")
         }
 
         # We also need the in/out transfers for the MultisigTxs
-        all_hashes = hashes_not_cached + [
-            multisig_tx.ethereum_tx_id for multisig_tx in multisig_txs.values()
+        all_hashes = id_not_cached + [
```

Comment:
Nitpick: Should this be renamed as well? (I mean, `all_ids/all_identifiers` instead of `all_hashes`?)

Reply:
Done!